### PR TITLE
network: use 'multiaddr' for validator interface

### PR DESCRIPTION
**Pull request by bmwill**
This patch converts the validator network interface to use and
understand multiaddr in place of using a (host, port) tuple.

### CODE REVIEW

**Comment by bmwill** [comment]
Note that we should not land this PR until https://github.com/MystenLabs/mysten-infra/pull/70 lands and we update this PR to point directly at a commit hosted in the https://github.com/MystenLabs/mysten-infra repo. This PR currently points at my local fork of the mysten-infra repo.

File: sui_core/src/consensus_adapter.rs [hunk 1]
```diff
@@ -93,18 +89,12 @@ impl ConsensusAdapter {
...
             .await
             .map_err(|e| SuiError::ConsensusConnectionBroken(e.to_string()))?;
 
-        let stream = FramedWrite::new(
-            stream,
-            LengthDelimitedCodec::builder()
-                .max_frame_length(buffer_size)
-                .new_codec(),
-        );
+        let stream = FramedWrite::new(stream, LengthDelimitedCodec::builder().new_codec());
```
Comment (gdanezis) [inline sui_core/src/consensus_adapter.rs:97]: Do we still need framed writes? Is not everything grpc across sui/narwhal?

**Review by gdanezis — APPROVED** [review]
Epic PR -- many thanks.

File: sui_core/src/consensus_adapter.rs (hunk 1 above)
Comment (bmwill) [reply sui_core/src/consensus_adapter.rs:97]: We still need to update the narwhal pointer in sui to a commit that includes the grpc changes we made. Once that's done we'll be able to drop the explicit framed streams.

File: sui/Cargo.toml [hunk 2]
```diff
@@ -33,6 +33,8 @@ serde-name = "0.2.0"
 dirs = "4.0.0"
 clap = { version = "3.1.14", features = ["derive"] }
 telemetry_subscribers = { git = "https://github.com/MystenLabs/mysten-infra", rev = "d2976a45420147ad821baae96e6fe4b12215f743" }
+multiaddr = "0.14.0"
+mysten-network = { git = "https://github.com/bmwill/mysten-infra.git", rev = "344471e3d3981585a2be7a5e25ea5fa8ba124a98" }
```
Comment (huitseeker) [inline sui/Cargo.toml:37]: Presumably, you could update telemetry-subscribers when you re-jig this with your merged mysten-infra revision.

**Review by huitseeker — APPROVED** [review]
This looks good, up to reconfiguration post mysten-infra merge.

File: sui/src/config/mod.rs [hunk 3]
```diff
@@ -105,6 +103,36 @@ fn socket_addr_from_hostport(host: &str, port: u16) -> SocketAddr {
...
+            if let Some(offset) = port_offset {
+                port + offset
+            } else {
+                port
+            }
+        }
+        _ => panic!("unexpected protocol"),
+    };
+
+    let mut addresses = format!("{host}:{port}")
```
Comment (velvia) [inline sui/src/config/mod.rs:128]: @longbowlu your retry logic PR will need to be merged with this right here

**Pull request merged**